### ISSUE TITLE
Wrong determination of version number in CHOLMOD

### ISSUE THREAD

**Issue by andreasnoack** [JuliaLang/julia#10348]
The version number is determined at compile time, which is not correct when Julia is from a repository where SuiteSparse/CHOLMOD could be different from the machine where the package is build.

**Comment by dmbates** [comment]
If the shared library provides a C function `cholmod_version` then that is used.  The compiled-in values are there only for old versions of the shared library.  It is quite likely that things will blow up if you need to use such a version.

**Comment by andreasnoack** [comment]
I was not precise here. When talking about compile, I was referring to the time when the system image is build. The relevant lines in cholmod.jl are
```julia
const chm_ver    = Array(Cint, 3)
if dlsym(dlopen("libcholmod"), :cholmod_version) != C_NULL
    ccall((:cholmod_version, :libcholmod), Cint, (Ptr{Cint},), chm_ver)
else
    ccall((:jl_cholmod_version, :libsuitesparse_wrapper), Cint, (Ptr{Cint},), chm_ver)
end
```
and I think this means that `chm_ver` is encoded when the system image is build, not when Julia is launched. Hence the version in `chm_ver` need not match the actual version of CHOLMOD that is actually loaded when Julia runs.

**Comment by andreasnoack** [comment]
Fixed by #10362
